### ISSUE TITLE
Deprecation warnings in Julia 0.7

### ISSUE THREAD

**Issue by galenlynch** [spalato/Destruct.jl#2]
```julia
(v0.7) pkg> test Destruct
   Testing Destruct
    Status `/tmp/tmpl0K5fY/Manifest.toml`
  [dde75ccc] Destruct v1.0.0
WARNING: Base.Test is deprecated, run `using Test` instead
  likely near /home/glynch/.julia/packages/Destruct/mKhWW/test/runtests.jl:2
WARNING: Base.AbstractRNG is deprecated: it has been moved to the standard library package `Random`.
Add `using Random` to your imports.
  likely near /home/glynch/.julia/packages/Destruct/mKhWW/test/runtests.jl:5
WARNING: Base.Complex128 is deprecated, use ComplexF64 instead.
  likely near /home/glynch/.julia/packages/Destruct/mKhWW/test/runtests.jl:6
Test Summary: | Pass  Total
NTuple        |   20     20
Test Summary: | Pass  Total
Tuple         |   48     48
Test Summary:     | Pass  Total
Tuple, Non square |   32     32
   Testing Destruct tests passed 
```

**Comment by spalato** [comment]
Thanks for the heads up. All errors are trivial, which is good to know.

I currently can't take the risk of a breaking upgrade past 0.6, and won't be able for a few weeks.

You can submit a pull request with the fixes if you wish.

**Comment by spalato** [comment]
Actually very simple to fix using Compat.

Travis stays this is fixed in Julia nightly, 1.0, 0.7.

Can you confirm?

**Comment by galenlynch** [comment]
Looks fine to me!

```julia
(v0.7) pkg> test Destruct
   Testing Destruct
 Resolving package versions...
    Status `/tmp/tmpzYLG3T/Manifest.toml`
  [34da2185] Compat v1.0.0 [`~/.julia/v0.6/Compat`]
  [dde75ccc] Destruct v1.0.0+ #master (https://github.com/spalato/Destruct.jl.git)
  [2a0f44e3] Base64  [`/usr/bin/../share/julia/stdlib/v0.7/Base64`]
  [ade2ca70] Dates  [`/usr/bin/../share/julia/stdlib/v0.7/Dates`]
  [8bb1440f] DelimitedFiles  [`/usr/bin/../share/julia/stdlib/v0.7/DelimitedFiles`]
  [8ba89e20] Distributed  [`/usr/bin/../share/julia/stdlib/v0.7/Distributed`]
  [b77e0a4c] InteractiveUtils  [`/usr/bin/../share/julia/stdlib/v0.7/InteractiveUtils`]
  [76f85450] LibGit2  [`/usr/bin/../share/julia/stdlib/v0.7/LibGit2`]
  [8f399da3] Libdl  [`/usr/bin/../share/julia/stdlib/v0.7/Libdl`]
  [37e2e46d] LinearAlgebra  [`/usr/bin/../share/julia/stdlib/v0.7/LinearAlgebra`]
  [56ddb016] Logging  [`/usr/bin/../share/julia/stdlib/v0.7/Logging`]
  [d6f4376e] Markdown  [`/usr/bin/../share/julia/stdlib/v0.7/Markdown`]
  [a63ad114] Mmap  [`/usr/bin/../share/julia/stdlib/v0.7/Mmap`]
  [44cfe95a] Pkg  [`/usr/bin/../share/julia/stdlib/v0.7/Pkg`]
  [de0858da] Printf  [`/usr/bin/../share/julia/stdlib/v0.7/Printf`]
  [3fa0cd96] REPL  [`/usr/bin/../share/julia/stdlib/v0.7/REPL`]
  [9a3f8284] Random  [`/usr/bin/../share/julia/stdlib/v0.7/Random`]
  [ea8e919c] SHA  [`/usr/bin/../share/julia/stdlib/v0.7/SHA`]
  [9e88b42a] Serialization  [`/usr/bin/../share/julia/stdlib/v0.7/Serialization`]
  [1a1011a3] SharedArrays  [`/usr/bin/../share/julia/stdlib/v0.7/SharedArrays`]
  [6462fe0b] Sockets  [`/usr/bin/../share/julia/stdlib/v0.7/Sockets`]
  [2f01184e] SparseArrays  [`/usr/bin/../share/julia/stdlib/v0.7/SparseArrays`]
  [10745b16] Statistics  [`/usr/bin/../share/julia/stdlib/v0.7/Statistics`]
  [8dfed614] Test  [`/usr/bin/../share/julia/stdlib/v0.7/Test`]
  [cf7118a7] UUIDs  [`/usr/bin/../share/julia/stdlib/v0.7/UUIDs`]
  [4ec0a83e] Unicode  [`/usr/bin/../share/julia/stdlib/v0.7/Unicode`]
Test Summary: | Pass  Total
NTuple        |   20     20
Test Summary: | Pass  Total
Tuple         |   48     48
Test Summary:     | Pass  Total
Tuple, Non square |   32     32
   Testing Destruct tests passed
```
